src/SOS/SOS.Extensions/HostServices.cs

```diff
@@ -106,20 +106,34 @@ static HostServices()
 
         private HostServices()
         {
-            _serviceProvider = new ServiceProvider();
-            _serviceProvider.AddService<IDiagnosticLoggingService>(DiagnosticLoggingService.Instance);
+            _serviceManager = new ServiceManager();
+            _serviceContainer = _serviceManager.CreateServiceContainer(ServiceScope.Global);
+            _serviceContainer.AddService<IServiceManager>(_serviceManager);
```

Comment:
The impl "ServiceManager" doesn't and shouldn't be exposed. Just the one method on IServiceManager "CreateServiceContainer". This is for extensions like DCR's extension when it creates IRuntime (or IModule, ITarget, etc) instances.  It doesn't reference the M.D.DS.Implementation just the interface assembly (M.D.DS). This is related to what I was saying in a previous comment about limited the "public" exposure with the interfaces.  Only hosts (dotnet-dump, lldb/windbg and VS) need the impl assembly or would ever need to create an impl of IServiceManager or IServiceContainer.  Actually since there are going to be so few "hosts" (4) out in the world and they can all reference our impl assembly no one will ever need to directly impl ServiceContainer or ServiceManager.